libraries/Filter/SlewLimiter2D.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  The SlewLimiter2D filter provides an actuator slew rate limiter for
+  PID controllers. It is used to reduce the P and D gains when the
+  filter detects that the P+D components are pushing the actuator
+  beyond the configured actuator rate limit. This can prevent
+  oscillations that are caused by the output actuation rate going
+  beyond the actuator maximum physical rate, which causes the
+  actuation demand and achieved rate to get out of phase.
+
+  this filter was originally written by Paul Riseborough for fixed
+  wing use. It was adapted for wider use in AC_PID by Andrew Tridgell
+ */
+#include "SlewLimiter2D.h"
+
+#define WINDOW_MS 300 // time in msec required for a half cycle of the slowest oscillation frequency expected
+#define MODIFIER_GAIN 1.5f // ratio of modifier reduction to slew rate exceedance ratio
+#define DERIVATIVE_CUTOFF_FREQ 25.0f
+
+SlewLimiter2D::SlewLimiter2D(const float &_slew_rate_max, const float &_slew_rate_tau) :
```

Comment:
should we call this as SlewDetector2D and not have the param?

Reply:
I've renamed the class to SlewCalculator2D and removed the parameter.